# Methods

DATA COLLECTED:

-   Nasal cytokines (Dessi)

-   Nasal cell flow (Simon & Jesus)

-   Nasal cell RNAseq (Fernando & Dessi)

-   Anti-CPS Baseline and D29 (Hugh)

-   MSD Baseline and D29

-   T cell ICS data ? Need to track down original data

EXPERIMENTS REMAINING:

-   Nasal Wash MPO -- neutrophil function

-   Re-challenge cohort - Anti-CPS Baseline and D29 and rechallenge

-   Serum OPK assays  6B(in-house) vs more serotypes (Goldblatt)

-   B cell phenotyping -- for senescence and memory B cell populations Same as for SPN3 with a few more markers

-   B cell ELISPOT  same protocol as SPN3

## Overall Experimental Design, Data and Analysis

### Experimental Designs and Data Measurements

To describe and identify the immunological mechanisms involved in the development of Spn carriage this project will analyze data from different cohorts of the EHPC consortium. The datasets selected comprise of 5 different cohorts, each with an experimental design allow to identify and describe the main components involved in the response to Spn infection, development of Spn carriage, its interactions with flu vaccines (TIV and LAIV), as well evaluate how these immunological responses behave between sex and age groups. Different experiments probe a specific level of the human nasal mucosa system such as gene expression, protein production and cytokine signalling, immune cell recruitment, Spn colonization and microbiota composition.

So far, 3 cohorts are fully completed: PILOT, LAIV1 and LAIV2. They are all constituted of healthy adult volunteers, with age ranging from 17 to 48 years old. The first cohort (PILOT) was designed to evaluate the immunological factors involved in the response to Spn infection, containing a relatively small set of 20 volunteers, when compared to other studies. The LAIV1 cohort, performed during the winter of 2015/2016, was developed to evaluate how influenza vaccines, such as trivalent inactivated (TIV) and live attenuated influenza (LAIV) vaccines, affects the development of carriage of 129 volunteers after being inoculated with Spn. The opposite biological question was asked in the following LAIV2 study, in which response to flu vaccines was evaluated regarding prior exposure to Spn. This cohort was performed in the winter of 2016/2017 with a total of 198 volunteers. The remaining two ongoing cohorts have also the same designs but probing the elderly population instead. The first study, namely Elderly, 82 volunteers from both genders and within 50 and 81 years old were inoculated with Spn and samples were collected at baseline and after inoculation, similarly to PILOT study. The last cohort, still in the development stage, will add the interaction with influenza vaccines to the previous design.

Different types of experiments were performed in all cohorts, measuring multiple levels of the human nasal mucosa: bulk RNA sequencing to represent the gene expression of basal cells, multiplex cytokine assays (Luminex) to measure 30 different immune factors involved in the immune system signalling. Cytometry was performed to estimate the number of cell types present at the nasal mucosa, as well as the density of Spn after inoculation. Virus serotyping and bulk RNA sequencing of the nasal mucosa was also carried aiming to measure nasal microbiota composition.

### Overall Data Types and Preprocessing

Only data without any preprocessing or transformation were used, for all cohorts and data types, except for data retrieved from public databases when additional information was required in a given step of the analysis (e.g., GMT files from Reactome database for pathway analysis).

Raw data from RNA-Seq experiments consists of fastq files and were preprocessed with a benchmark pipeline: samples were aligned using the STAR (Dobin et al., 2013)⁠ software, transcripts counts calculated with featureCounts (Liao, Smyth, & Shi, 2014)⁠ software and, with personalized scripts in bash and R language, samples of a given cohort were merge in a single table, where each row represent a transcript and each column a sample. Both STAR and featureCounts steps were supplied with the Hg38 reference (Herrero et al., 2016; Ruffier et al., 2017)⁠⁠ genome and features, represented by Ensembl identifiers (EnsemblID, Aken et al., 2016), were parsed so that the EnsemblID\`s version is omitted. Quality control checks were performed between each step of the preprocessing procedure, with fastQC (A. & Bittencourt a, 2010)⁠ on fastq files before alignment and MultiQC (Ewels, Magnusson, Lundin, & Käller, 2016)⁠ for all steps with no samples removed due to low quality. Counts data were normalized either with trimmed means of M-values (TMM, Smid et al., 2018)⁠ or variance stabilizing transformation (vst) (Zwiener, Frisch, & Binder, 2014)⁠with the DESeq2 (Love, Huber, & Anders, 2014)⁠ package from Bioconductor (Ihaka & Gentleman, 1996)⁠⁠, when necessary.

For the remaining data types, raw data were retrieved in standard comma/tab-separated formats (e.g. CSV), containing tables representing either absolute measurements, proportions or mean levels of biological variables (e.g. EGF, IL-10 for Luminex or T-Cell Percentage for flow cytometry data) in the columns, for a given volunteer and time-point in the rows. Raw values were shifted by a constant value (all values are added by the absolute minimum value of the dataset) to avoid negative/zero value and then log2 transformed. Depending on the analysis step, each variable could be further scaled by the Z-Score transformation. Variables or features with more than 25% of missing values were excluded, with the remaining missing data imputed through an unsupervised approach algorithm implemented in the missForest (Stekhoven & Buhlmann, 2012)⁠ package.

Each dataset was further inspected for data quality, outliers and technical artefacts or batch effects, and overall variable distribution. This was done with usual exploratory data analysis (EDA), such as dimensionality reduction algorithms (e.g. Principal Component Analysis (PCA) and Independent PCA (IPCA)), clustering techniques (e.g. Hierarchical Clustering (HC)) and visualization tools provided by R packages.

### Biological Pathways

With the results obtained from the analysis of DE and co-expression analysis of RNA-Seq data, we get new sets of variables that could potentially have some biological meaning, as explained in the previous section. For instance, after the identification of DEGs obtained in a certain comparison, we will possibly have two lists of genes that are up- and down-regulated. The same is true for the co-expression analysis, in which we will get a different list of genes for each co-expression module identified during the analysis. All those lists should then be further explored through enrichment analysis to identify which biological pathways are most probably related in order to uncover the underlying molecular mechanisms.

To identify which biological pathways could potentially be perturbed in each comparison of a DE analysis, gene set enrichment analysis (A. Subramanian et al., 2005)⁠ was performed upon the ranked log2FC values. The Reactome (Croft et al., 2011)⁠ gene set (GMT file) from Enrichr (Chen et al., 2013; Kuleshov et al., 2016)⁠ database was retrieved from Enrichr database. Also, the EnsemblIDs of each transcript was translated into Gene Symbol IDs with the biomaRt (Durinck, Spellman, Birney, & Huber, 2009)⁠ package so the transcripts could be matched between the datasets and Reactome gene set. The actual enrichment was performed with the fgsea function from fgsea (Sergushichev, 2016)⁠ package, with all parameters set as default, except for the number of permutations (nperm) that was set to 1000. The enrichment results of each comparison were merged with all comparison in each analysis, and all pathways with padj \< 0.05 in at least one of the comparisons were selected to further exploratory analysis. To further explore such selected pathways, common pathways for all groups or time-points were displayed in correlation plots, from corrplot (Wei & Simko, 2017)⁠ package, with pathways represented in rows, comparison represented in columns and enrichment (NES) values as circles, coloured in a blue-to-red gradient, representing negative to positive perturbation respectively.

## Proposed Solution

### Overall Data Analysis

#### Identification of Perturbed Features

To first answer the most basic question, "which biological variable(s) are possibly related to Spn inoculation?", features that suffered any perturbation were detected through statistical methods which compare their sample distribution after the inoculation to its baseline levels. In RNA-Seq data, this analysis is performed with both DESeq2 and edgeR (Robinson, McCarthy, & Smyth, 2009)⁠ R packages and it is known as differential expression analysis (DEA). This perturbation is represented by the ratio of distribution means and displayed as log2 fold-change (log2FC) values. Statistical confidence is also estimated, represented by the p-values (pval), and further adjusted (padj) for multiple comparisons. Deferentially expressed genes (DEG) can be selected by setting thresholds for both log2FCs and p-values, and classified in up-, down-regulated or unperturbed according to its log2FCs direction. In this study DEGs were defined according to the following thresholds: among the genes with padj \< 0.01 the ones with log2FC \> 0 are classified as UP, log2FC \< 0 classified as DOWN and the remaining as UNCHANGED.

Similar statistical tests were applied to the remaining data types using R core functions. These tests can be parametric (e.g. T-Test) or not (e.g. Mann-Whitney-Wilcoxon (MWW test) and are applied accordingly if most of the variables in a given dataset follow a normal distribution or not, respectively. P-values obtained from these tests were corrected for multiple comparisons with Bonferroni method (Armstrong, 2014)⁠ and variables with *padj* below a threshold of 0.01 were selected for further analysis.

#### Identification of Transcriptional Programs and Variables Cluster with Similar Patterns

Beyond identifying perturbed variables, one might want to discover groups of variables that have similar patterns throughout the time-points or between conditions. One example of this type of analysis is co-expression analysis, usually performed with transcriptomic data. Such analysis assumes that genes with similar behaviour or pattern of expression could be biological entities that are working together, potentially representing a biological program and perhaps being orchestrated by common regulators. For example, a given set of highly correlated genes could indicate that they are all member of the apoptosis mechanism and might even be regulated by the same transcriptional factor, long non-coding RNA (lincRNAs) or micro RNA (miRNA). This type of analysis is very powerful to identify major patterns in the data, being them transcriptomic data or not.

In the case of RNA-Seq data, co-expression analysis were performed using the CEMiTool (Russo et al., 2018)⁠ package, which allow the identification of co-expression modules (MOD), visualize their overall expression along samples, identify most probable biological pathways that a given MOD could be related to and also analyze how each module are behaving in each condition or time-point, in other words, identify if the given module is up- or down-regulated in each condition/time-point by analyzing if the majority of its constituents (genes/transcripts) are highly enriched among the highly positively or negatively expressed genes.

The same basic idea was also applied to the other types of data in order to identify possible relationship patterns among the biological variables. For example, identifying sets of cytokines (Luminex) or immune cell types (flow cytometry) that are highly correlated between each other, either across time-points or groups (e.g. Spn groups, vaccine type, etc.). This sets of highly correlated variables were determined through the calculation of pairwise correlations (Spearman, Rho) for all variables in each dataset. These pairwise correlations are represented in the form of a correlation matrix, with the number of rows and columns equal to the total number of variables present in the dataset. To define if two variables are connected or not, discrete values of 0s and 1s were used to represent if a connection exists, respectively. Values of 1 were assigned if their absolute correlation values were \|Rho\| \>= 0.7, 0 otherwise. This discrete matrix, also known as an adjacency matrix, was used to create a graph, where the nodes represent biological variables (e.g. cytokines, cell type) and can be connected (edges) only if its adjacency value is equal to 1. Finally, variables were grouped into modules through the Louvain (Newman, 2006; Traag, Waltman, & van Eck, 2019)⁠ clustering method, a commonly used community detection algorithm and available in igraph (Csárdi & Nepusz, n.d.)⁠ package.

## Transcriptomics Analysis Pipeline

### Quality Control

-   Density plot

-   Clustering

-   MA plot

    -   microarray - affycoretools::maplot() - <https://rdrr.io/bioc/t/man/maplot.html>

    -   rnaseq - DESeq2::plotMA(dds)

-   Distribution (Normality) test

-   PVCA

-   MDP

-   RLE

-   Multidimensional scaling (MDS)

-   Bi-clustering

-   PCA

-   IPVCA

-   mixOmics

### Biomarkers

-   DEG Analysis

    -   Microarray

        -   distribution assumptions evaluation

-   RNA-Seq

    -   DESeq2

    -   edgeR

    -   limma

-   DE analysis:

    -   limma

    -   t-test

    -   ANOVA

    -   ranking methods

        -   see other methods and approaches

    -   *p-value* distribution analysis

    -   pathway stabilization for DEGs thresholding

-   Bi-clustering

    -   <https://rpubs.com/crazyhottommy/PCA_MDS>

### Pathway Analysis

-   ORA

    -   Network Analyst

    -   Enrichr

    -   StringR

-   GSEA

-   IPA

### Pipeline

INPUT: - Raw expression data - RNA-Seq: counts - Microarray: .CEL, raw expression table (Illumina) - phenodata

    Parameters:
        - split_proportion = c(#train, #test, #validation)
        - response_var = ""
        - preliminary_filter = c(T,F)

OUTPUT: - biomarkers list - my performance assessment - plots

ALGORITHM - Read data - test tables equivalence - split data in train, test and validation sets - (Optional) preliminary filter (e.g, low variance, low mean) - Biomarker detecion - Biomarkers: biomarker list extraction - Biomarkers performance test in train/test dataset - Biomarkers performance test in validation dataset - plots

### 

## Network Analysis

### Co-expression Analysis

-   WGCNA / CEMiTool

<!-- -->

-   ARACNE

-   Topological analysis
