# Resources

## The Human Cell Atlas

Introduction

Between 1838 and 1855, Schleiden, Schwann, Remak, Virchow and others crystalized an elegant Cell Theory:

1. all organisms are composed of one or more cells;

2. that cells are the basic unit of structure and function in life;

3. and that all cells are derived from pre-existing cells;

Human physiology emerges from normal cellular functions and interactions. Human disease entails the disruption of these processes an may involve aberrant cell types and states. Genetic variants that contribute to disease typically manifest their action through impact in a particular cell types.

At a conceptual level, one challenge is that we lack a rigorous definition of what we mean by the intuitive terms "cell type" and "cell state." Cell type often implies a notion of persistence (e.g., being a hepatic stellate cell or a cerebellar Purkinje cell), while cell state often refers to more transient properties (e.g., being in the G1 phase of the cell cycle or experiencing nutrient deprivation). But, the boundaries between these concepts can be blurred, because cells change over time in ways that are far from fully understood.

What is the Human Atlas?

At its most basic level, the Human Cell Atlas must include a comprehensive

reference catalog of all human cells based on their stable properties and transient features,

as well as their locations and abundances. Yet, an atlas is more than just a catalog: it is a

map that aims to show the relationships among its elements. By doing so, it can

sometimes reveal fundamental processes---akin to how the atlas of Earth suggested

continental drift through the correspondence of coastlines.

To be useful, an atlas must also be an abstraction---comprehensively representing

certain features, while ignoring others.

A natural solution would be to describe each human

cell by a defined set of molecular markers. For example, one might describe each cell by

the expression level of each of the \~20,000 human protein-coding genes---that is, each cell

would be represented as a point in \~20,000-dimensional space. Of course, the set of

markers could be expanded to include the expression levels of non-coding genes, the

levels of the alternatively spliced forms of each transcript, the chromatin state of every

promoter and enhancer, and the levels of each protein or each post-translationally

modified form of each protein.

The imaginary Ultimate Human Cell Atlas would represents all conceivable markers in:

1. every cell in a person's body

2. every cell's spatial position (by adding three dimensions for the body axes)

3. every cell at every moment of a person's lifetime (by adding another dimension for time relating the cells by a lineage)

4. the superimposition of such atlases from every human being, annotated according to differences in health, genotype, lifestyle and environmental exposure

A cell "type" might be defined as a region or a probability distribution --- either in the full-dimensional space or in a projection onto a lower-dimensional space that reflects salient features.

Cell types follow a highly stereotyped spatial patterns.

Histology: Cell neighborhood and position

Histology examines the spatial position of cells and molecules within tissues.

Development: transitions to differentiated cell types

Cells arrive at their final differentiated cell types through partly asynchronous

branching pathways of development, which are driven by and reflected in molecular

changes, especially gene-expression patterns

a cell may be following multiple dynamic paths simultaneously---for example, differentiation, the cell cycle, and pathogen response --- that may affect each other.

Physiology	and	homeostasis:	cycles,	transient	responses	and	plastic	states

	cells	 are	 constantly	 undergoing	multiple dynamic	 processes	 of	 physiological	 change and	 homeostatic	 regulation:

		cyclical	processes,	such	as	the	cell	cycle	and	circadian	rhythms;

		transient	 responses to	 diverse	 factors,	 from	 nutrients	 and	 microbes to	mechanical forces	and	tissue	damage;	

		plastic	states that	can	be	stably	maintained	over	longer	 time	 scales,	but can	change	in	 response to	new	environmental	cues.		(The	precise	boundary between	plastic	states	and	cell	types,	it	must	be	noted,	remains	to	be	clarified.)

	The	molecular	phenotype	of	a	cell	reflects	a	superposition	of	these	various	processes	and	their	interactions

Disease: Cells	and	cellular ecosystems

	disease,	 which	invariably involves	 disruption	 of	 normal	 cellular	 functions,	 interactions,	 proportions,	 or	ecosystems.	

	Single-cell	information	across	many patients	will	allow	us	to	learn	about	how	cell	

proportions	 and	 states	 vary	 and	 how	 this	 variation	 correlates	 with	 genome	 variants,	

disease	 course	 and	 treatment	 response

## reproducible data analysis

	[OK] <https://resources.rstudio.com/rstudio-conf-2019/a-guide-to-modern-reproducible-data-science-with-r>

	

	=\> <https://ropensci.github.io/reproducibility-guide/>

	<http://grunwaldlab.github.io/Reproducible-science-in-R/>

		<https://github.com/grunwaldlab/Reproducible-science-in-R>

	<https://rpubs.com/minebocek/user2017-ors>

	<https://www.r-bloggers.com/preview-my-new-book-introduction-to-reproducible-science-in-r/>

	<https://datacarpentry.org/rr-workshop/>

	<https://rviews.rstudio.com/2018/01/18/package-management-for-reproducible-r-code/>

	<https://cartesianfaith.files.wordpress.com/2018/11/rowe-introduction-to-reproducible-science-part-i-20181113.pdf>

	<https://www.reconlearn.org/post/reproducibility.html>

	<http://www.geo.uzh.ch/microsite/reproducible_research/post/rr-rstudio-git/>

	<https://swcarpentry.github.io/r-novice-gapminder/02-project-intro/>

	<https://www.crcpress.com/Reproducible-Research-with-R-and-R-Studio/Gandrud/p/book/9781498715379>

	<https://www.tandfonline.com/doi/abs/10.1080/00031305.2017.1375986?journalCode=utas20>

	<https://www.tandfonline.com/doi/full/10.1080/00031305.2017.1375989?src=recsys>

	<https://github.com/ropenscilabs/checkers>

	<https://github.com/benmarwick/onboarding-reproducible-compendia>

	<https://github.com/research-compendium/ten-simple-rules>

	<https://github.com/research-compendium/how-to-read-a-research-compendium>

	<https://mybinder.org>

	<https://ropenscilabs.github.io/drake-manual/>

	<https://peerj.com/preprints/3159/>

	<https://happygitwithr.com/big-picture.html>

	<https://ropenscilabs.github.io/drake-manual/>

	<https://plotly-r.com/introduction.html>

<https://ironholds.org/projects/r_shiny/index.html>

<https://bioinformatics.ca/workshops/2016-exploratory-analysis-biological-data-using-r/>

<https://r4ds.had.co.nz/exploratory-data-analysis.html>

<https://ropensci.github.io/reproducibility-guide/>

<https://www.earthdatascience.org/tags/reproducible-science-and-programming/RStudio/>

<http://ohi-science.org/data-science-training/>

<http://grunwaldlab.github.io/Reproducible-science-in-R>

<https://www.jove.com/video/50115/experimental-human-pneumococcal-carriage>

analysis report

	rmarkdown

		<https://rmarkdown.rstudio.com>

		<https://www.rstudio.com/wp-content/uploads/2015/03/rmarkdown-reference.pdf>

	LaTex

		<https://wch.github.io/latexsheet/>

programming

	R

		<https://www.rstudio.com/wp-content/uploads/2016/02/advancedR.pdf>

exploratory data analysis

	<https://bookdown.org/rdpeng/exdata/preface.html>

	<https://towardsdatascience.com/simple-fast-exploratory-data-analysis-in-r-with-dataexplorer-package-e055348d9619>

	<https://blog.datascienceheroes.com/exploratory-data-analysis-in-r-intro/>

	<https://www.analyticsvidhya.com/blog/2016/02/complete-tutorial-learn-data-science-scratch/>

	<https://r4ds.had.co.nz/transform.html>

	<https://www.youtube.com/watch?v=MCJD5iJjr7Y>

	<http://www.sthda.com/english/wiki/print.php?id=202>

	<http://www.sthda.com/english/wiki/factoextra-r-package-easy-multivariate-data-analyses-and-elegant-visualization>

	<https://www.youtube.com/watch?v=aiqGyLGrQ48&list=PLnZgp6epRBbTsZEFXi_p6W48HhNyqwxIu&index=2&t=0s>

data science

	<https://github.com/rstudio/Intro/tree/master/slides>

	<https://www.oreilly.com/library/view/introduction-to-data/9781491915028/video192702.html>

bioinformatics

	<https://bioinformatics.ca/workshops/workshops-2018/>

	<https://www.ebi.ac.uk/training/online/course/bioinformatics-terrified-0>

	<https://www.ebi.ac.uk/training/online/course/bioinformatics-sequence-analysis-web-services-job-dispatcher>

	<https://www.ebi.ac.uk/training/online/course/bringing-data-life-data-management-biomolecular-sciences>

	<https://www.ebi.ac.uk/training/online/course/challenges-and-opportunities-virtual-research-environments>

systems biology

	<https://www.ebi.ac.uk/training/events/2019/systems-biology-large-datasets-biological-insight>

metagenomics

	<https://www.ebi.ac.uk/training/online/course/ebi-metagenomics-analysing-and-exploring-metagenomics-data>

	<https://www.ebi.ac.uk/training/online/course/analysing-and-visualising-microbiome-derived-datasets-using-mgnify-web-api>

data analysis certificates

	<https://www.colorado.edu/earthlab/earth-data-analytics-foundations-professional-certificate?utm_source=eds&utm_medium=website&utm_campaign=certificate-2018>

TIMESERIES

	<https://subscription.packtpub.com/book/big_data_and_business_intelligence/9781786462411>

## Databases

Genes associated with a disease:

	<http://www.disgenet.org/>

	<https://www.malacards.org/>

Genes associated with a tissue type:
