<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Definitions | Ageing and Sex Differences in Human Pneumococcal Carriage</title>
  <meta name="description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Definitions | Ageing and Sex Differences in Human Pneumococcal Carriage" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Definitions | Ageing and Sex Differences in Human Pneumococcal Carriage" />
  
  <meta name="twitter:description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  

<meta name="author" content="Fernando Marcon Passos" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Sex and Ageing Effects on Pneumococcal Carriage</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>a</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Background Knowledge</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#pneumonia-and-the-pneumococcal-diseases"><i class="fa fa-check"></i><b>1.1</b> Pneumonia and the Pneumococcal diseases</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#streptococcus-pneumoniae"><i class="fa fa-check"></i><b>1.2</b> Streptococcus pneumoniae</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#human-protection-to-pneumococcal-diseases"><i class="fa fa-check"></i><b>1.3</b> Human Protection to Pneumococcal Diseases</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#the-effects-of-sex-and-ageing-on-human-immunity"><i class="fa fa-check"></i><b>1.4</b> The Effects of Sex and Ageing on Human Immunity</a></li>
<li class="chapter" data-level="1.5" data-path="intro.html"><a href="intro.html#the-experimental-human-pneumococcal-carriage-model"><i class="fa fa-check"></i><b>1.5</b> The Experimental Human Pneumococcal Carriage Model</a></li>
<li class="chapter" data-level="1.6" data-path="intro.html"><a href="intro.html#systems-vaccinology"><i class="fa fa-check"></i><b>1.6</b> Systems Vaccinology</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="research-questions.html"><a href="research-questions.html"><i class="fa fa-check"></i><b>2</b> Research Questions</a></li>
<li class="chapter" data-level="3" data-path="project-plan.html"><a href="project-plan.html"><i class="fa fa-check"></i><b>3</b> Project Plan</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a>
<ul>
<li class="chapter" data-level="4.1" data-path="methods.html"><a href="methods.html#overall-experimental-design-data-and-analysis"><i class="fa fa-check"></i><b>4.1</b> Overall Experimental Design, Data and Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="methods.html"><a href="methods.html#experimental-designs-and-data-measurements"><i class="fa fa-check"></i><b>4.1.1</b> Experimental Designs and Data Measurements</a></li>
<li class="chapter" data-level="4.1.2" data-path="methods.html"><a href="methods.html#overall-data-types-and-preprocessing"><i class="fa fa-check"></i><b>4.1.2</b> Overall Data Types and Preprocessing</a></li>
<li class="chapter" data-level="4.1.3" data-path="methods.html"><a href="methods.html#biological-pathways"><i class="fa fa-check"></i><b>4.1.3</b> Biological Pathways</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="methods.html"><a href="methods.html#proposed-solution"><i class="fa fa-check"></i><b>4.2</b> Proposed Solution</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="methods.html"><a href="methods.html#overall-data-analysis"><i class="fa fa-check"></i><b>4.2.1</b> Overall Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="methods.html"><a href="methods.html#transcriptomics-analysis-pipeline"><i class="fa fa-check"></i><b>4.3</b> Transcriptomics Analysis Pipeline</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="methods.html"><a href="methods.html#quality-control"><i class="fa fa-check"></i><b>4.3.1</b> Quality Control</a></li>
<li class="chapter" data-level="4.3.2" data-path="methods.html"><a href="methods.html#biomarkers"><i class="fa fa-check"></i><b>4.3.2</b> Biomarkers</a></li>
<li class="chapter" data-level="4.3.3" data-path="methods.html"><a href="methods.html#pathway-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Pathway Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="methods.html"><a href="methods.html#pipeline"><i class="fa fa-check"></i><b>4.3.4</b> Pipeline</a></li>
<li class="chapter" data-level="4.3.5" data-path="methods.html"><a href="methods.html#section"><i class="fa fa-check"></i><b>4.3.5</b> </a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="methods.html"><a href="methods.html#network-analysis"><i class="fa fa-check"></i><b>4.4</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="methods.html"><a href="methods.html#co-expression-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Co-expression Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><a href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><i class="fa fa-check"></i><b>5</b> Data Driven Analysis of Susceptibility to Pneumococcal Carriage</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><a href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html#q1-is-developing-carriage-random"><i class="fa fa-check"></i><b>5.1</b> Q1: Is developing carriage random?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="definitions.html"><a href="definitions.html"><i class="fa fa-check"></i><b>6</b> Definitions</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="essential-literature.html"><a href="essential-literature.html"><i class="fa fa-check"></i><b>A</b> Essential Literature</a>
<ul>
<li class="chapter" data-level="A.1" data-path="essential-literature.html"><a href="essential-literature.html#thesis"><i class="fa fa-check"></i><b>A.1</b> Thesis</a></li>
<li class="chapter" data-level="A.2" data-path="essential-literature.html"><a href="essential-literature.html#articles"><i class="fa fa-check"></i><b>A.2</b> Articles</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="essential-literature.html"><a href="essential-literature.html#pneumococcal-diseases"><i class="fa fa-check"></i><b>A.2.1</b> Pneumococcal Diseases</a></li>
<li class="chapter" data-level="A.2.2" data-path="essential-literature.html"><a href="essential-literature.html#sex-differences"><i class="fa fa-check"></i><b>A.2.2</b> Sex Differences</a></li>
<li class="chapter" data-level="A.2.3" data-path="essential-literature.html"><a href="essential-literature.html#ageing"><i class="fa fa-check"></i><b>A.2.3</b> Ageing</a></li>
<li class="chapter" data-level="A.2.4" data-path="essential-literature.html"><a href="essential-literature.html#systems-biology"><i class="fa fa-check"></i><b>A.2.4</b> Systems Biology</a></li>
<li class="chapter" data-level="A.2.5" data-path="essential-literature.html"><a href="essential-literature.html#bioinformatics-and-methodologies"><i class="fa fa-check"></i><b>A.2.5</b> Bioinformatics and Methodologies</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html"><i class="fa fa-check"></i><b>B</b> Pneumococcal Diseases and Immunity</a>
<ul>
<li class="chapter" data-level="B.1" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html#controlled-human-infection-and-rechallenge-with-streptococcus-pneumoniae-reveals-the-protecticve-efficacy-of-carriage-in-healty-adults"><i class="fa fa-check"></i><b>B.1</b> Controlled Human Infection and Rechallenge with Streptococcus pneumoniae Reveals the Protecticve Efficacy of Carriage in Healty Adults</a></li>
<li class="chapter" data-level="B.2" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html#inflammation-induced-by-influenza-virus-impairs-innate-control-of-human-pneumococcal-carriage"><i class="fa fa-check"></i><b>B.2</b> Inflammation induced by influenza virus impairs innate control of human pneumococcal carriage</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="sex-differences-1.html"><a href="sex-differences-1.html"><i class="fa fa-check"></i><b>C</b> Sex Differences</a></li>
<li class="chapter" data-level="D" data-path="ageing-1.html"><a href="ageing-1.html"><i class="fa fa-check"></i><b>D</b> Ageing</a>
<ul>
<li class="chapter" data-level="D.1" data-path="ageing-1.html"><a href="ageing-1.html#the-hallmarks-of-aging"><i class="fa fa-check"></i><b>D.1</b> The Hallmarks of Aging</a></li>
<li class="chapter" data-level="D.2" data-path="ageing-1.html"><a href="ageing-1.html#therapies"><i class="fa fa-check"></i><b>D.2</b> Therapies</a></li>
<li class="chapter" data-level="D.3" data-path="ageing-1.html"><a href="ageing-1.html#senescent-cells"><i class="fa fa-check"></i><b>D.3</b> Senescent Cells</a></li>
<li class="chapter" data-level="D.4" data-path="ageing-1.html"><a href="ageing-1.html#the-role-of-senescent-cells-in-aging"><i class="fa fa-check"></i><b>D.4</b> The Role of Senescent Cells in Aging</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="bioinformatics.html"><a href="bioinformatics.html"><i class="fa fa-check"></i><b>E</b> Bioinformatics</a>
<ul>
<li class="chapter" data-level="E.1" data-path="bioinformatics.html"><a href="bioinformatics.html#bioinformatics-essentials"><i class="fa fa-check"></i><b>E.1</b> Bioinformatics Essentials:</a></li>
<li class="chapter" data-level="E.2" data-path="bioinformatics.html"><a href="bioinformatics.html#differential-expression-analysis"><i class="fa fa-check"></i><b>E.2</b> Differential Expression Analysis</a>
<ul>
<li class="chapter" data-level="E.2.1" data-path="bioinformatics.html"><a href="bioinformatics.html#empirical-assessment-of-analysis-workflows-for-differential-expression-analysis-of-human-samples-using-rna-seq"><i class="fa fa-check"></i><b>E.2.1</b> Empirical assessment of analysis workflows for differential expression analysis of human samples using RNA-Seq</a></li>
</ul></li>
<li class="chapter" data-level="E.3" data-path="bioinformatics.html"><a href="bioinformatics.html#network-analysis-1"><i class="fa fa-check"></i><b>E.3</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="E.3.1" data-path="bioinformatics.html"><a href="bioinformatics.html#genemania-a-real-time-multiple-association-network-integration-algorithm-for-predicting-gene-function"><i class="fa fa-check"></i><b>E.3.1</b> 2008 - GeneMANIA: a real-time multiple association network integration algorithm for predicting gene function</a></li>
<li class="chapter" data-level="E.3.2" data-path="bioinformatics.html"><a href="bioinformatics.html#the-genemania-prediction-server-biological-network-integration-for-gene-prioritization-and-predicting-gene-function"><i class="fa fa-check"></i><b>E.3.2</b> 2010 - The GeneMANIA prediction server: biological network integration for gene prioritization and predicting gene function</a></li>
</ul></li>
<li class="chapter" data-level="E.4" data-path="bioinformatics.html"><a href="bioinformatics.html#multidimensional-scaling"><i class="fa fa-check"></i><b>E.4</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="E.5" data-path="bioinformatics.html"><a href="bioinformatics.html#next-generation-sequencing"><i class="fa fa-check"></i><b>E.5</b> Next-Generation Sequencing</a>
<ul>
<li class="chapter" data-level="E.5.1" data-path="bioinformatics.html"><a href="bioinformatics.html#rna-seq-analysis"><i class="fa fa-check"></i><b>E.5.1</b> RNA-Seq Analysis</a></li>
<li class="chapter" data-level="E.5.2" data-path="bioinformatics.html"><a href="bioinformatics.html#microbiome-data-analysis"><i class="fa fa-check"></i><b>E.5.2</b> microbiome data analysis</a></li>
</ul></li>
<li class="chapter" data-level="E.6" data-path="bioinformatics.html"><a href="bioinformatics.html#omics"><i class="fa fa-check"></i><b>E.6</b> OMICS</a>
<ul>
<li class="chapter" data-level="E.6.1" data-path="bioinformatics.html"><a href="bioinformatics.html#integromics-an-r-package-to-unravel-relationships-between-two-omics-datasets"><i class="fa fa-check"></i><b>E.6.1</b> integrOmics an R package to unravel relationships between two omics datasets</a></li>
<li class="chapter" data-level="E.6.2" data-path="bioinformatics.html"><a href="bioinformatics.html#visualising-associations-between-paired-omics-data-sets"><i class="fa fa-check"></i><b>E.6.2</b> Visualising associations between paired “omics” data sets</a></li>
</ul></li>
<li class="chapter" data-level="E.7" data-path="bioinformatics.html"><a href="bioinformatics.html#principal-component-analysis"><i class="fa fa-check"></i><b>E.7</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="E.8" data-path="bioinformatics.html"><a href="bioinformatics.html#mixomics"><i class="fa fa-check"></i><b>E.8</b> MixOmics</a>
<ul>
<li class="chapter" data-level="E.8.1" data-path="bioinformatics.html"><a href="bioinformatics.html#diablo"><i class="fa fa-check"></i><b>E.8.1</b> DIABLO</a></li>
<li class="chapter" data-level="E.8.2" data-path="bioinformatics.html"><a href="bioinformatics.html#exploratory-data-analysis-with-mixomics"><i class="fa fa-check"></i><b>E.8.2</b> EXPLORATORY DATA ANALYSIS WITH mixOMICS</a></li>
<li class="chapter" data-level="E.8.3" data-path="bioinformatics.html"><a href="bioinformatics.html#independent-component-analysis-ica"><i class="fa fa-check"></i><b>E.8.3</b> - Independent Component Analysis (ICA)</a></li>
<li class="chapter" data-level="E.8.4" data-path="bioinformatics.html"><a href="bioinformatics.html#independent-principal-component-analysis-ipca"><i class="fa fa-check"></i><b>E.8.4</b> Independent Principal Component Analysis (IPCA)</a></li>
<li class="chapter" data-level="E.8.5" data-path="bioinformatics.html"><a href="bioinformatics.html#mixomics---an-r-package-for-omics-feature-selection-and-multiple-data-integration_2017"><i class="fa fa-check"></i><b>E.8.5</b> mixOmics - An R package for omics feature selection and multiple data integration_2017</a></li>
<li class="chapter" data-level="E.8.6" data-path="bioinformatics.html"><a href="bioinformatics.html#partial-least-squares-regression-pls-regression"><i class="fa fa-check"></i><b>E.8.6</b> Partial Least Squares regression (PLS regression)</a></li>
<li class="chapter" data-level="E.8.7" data-path="bioinformatics.html"><a href="bioinformatics.html#variable-selection-for-generalized-canonical-correlation-analysis"><i class="fa fa-check"></i><b>E.8.7</b> Variable selection for generalized canonical correlation analysis</a></li>
<li class="chapter" data-level="E.8.8" data-path="bioinformatics.html"><a href="bioinformatics.html#unravelling-omics-data-with-the-mixomics-r-package-ppt"><i class="fa fa-check"></i><b>E.8.8</b> Unravelling “omics” data with the mixOmics R package (ppt)</a></li>
</ul></li>
<li class="chapter" data-level="E.9" data-path="bioinformatics.html"><a href="bioinformatics.html#normalization"><i class="fa fa-check"></i><b>E.9</b> Normalization</a>
<ul>
<li class="chapter" data-level="E.9.1" data-path="bioinformatics.html"><a href="bioinformatics.html#the-universal-expression-code-upc-algorithm-consists-of-two-main-steps"><i class="fa fa-check"></i><b>E.9.1</b> The Universal exPression Code (UPC) algorithm consists of two main steps:</a></li>
</ul></li>
<li class="chapter" data-level="E.10" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-analysis"><i class="fa fa-check"></i><b>E.10</b> Time-Series Analysis</a>
<ul>
<li class="chapter" data-level="E.10.1" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-clustering"><i class="fa fa-check"></i><b>E.10.1</b> Time-Series Clustering</a></li>
<li class="chapter" data-level="E.10.2" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-options"><i class="fa fa-check"></i><b>E.10.2</b> Time-Series Options</a></li>
</ul></li>
<li class="chapter" data-level="E.11" data-path="bioinformatics.html"><a href="bioinformatics.html#survival-analysis"><i class="fa fa-check"></i><b>E.11</b> Survival analysis</a></li>
<li class="chapter" data-level="E.12" data-path="bioinformatics.html"><a href="bioinformatics.html#data-integration"><i class="fa fa-check"></i><b>E.12</b> Data Integration</a>
<ul>
<li class="chapter" data-level="E.12.1" data-path="bioinformatics.html"><a href="bioinformatics.html#sparse-canonical-methods-for-biological-data-integration-application-to-a-cross-platform-study"><i class="fa fa-check"></i><b>E.12.1</b> Sparse canonical methods for biological data integration: application to a cross-platform study</a></li>
</ul></li>
<li class="chapter" data-level="E.13" data-path="bioinformatics.html"><a href="bioinformatics.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>E.13</b> Single-Cell RNA-Seq</a>
<ul>
<li class="chapter" data-level="E.13.1" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis"><i class="fa fa-check"></i><b>E.13.1</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="E.14" data-path="bioinformatics.html"><a href="bioinformatics.html#reproducible-data-science"><i class="fa fa-check"></i><b>E.14</b> Reproducible Data Science</a></li>
<li class="chapter" data-level="E.15" data-path="bioinformatics.html"><a href="bioinformatics.html#machine-learning-vs-de-analysis"><i class="fa fa-check"></i><b>E.15</b> machine learning vs DE Analysis</a></li>
<li class="chapter" data-level="E.16" data-path="bioinformatics.html"><a href="bioinformatics.html#dealing-with-missing-data"><i class="fa fa-check"></i><b>E.16</b> Dealing with Missing Data</a></li>
<li class="chapter" data-level="E.17" data-path="bioinformatics.html"><a href="bioinformatics.html#feature-selection"><i class="fa fa-check"></i><b>E.17</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="E.17.1" data-path="bioinformatics.html"><a href="bioinformatics.html#features-and-feature-engineering"><i class="fa fa-check"></i><b>E.17.1</b> Features and feature engineering</a></li>
</ul></li>
<li class="chapter" data-level="E.18" data-path="bioinformatics.html"><a href="bioinformatics.html#exploratory-data-anaysis-eda"><i class="fa fa-check"></i><b>E.18</b> Exploratory Data Anaysis (EDA)</a>
<ul>
<li class="chapter" data-level="E.18.1" data-path="bioinformatics.html"><a href="bioinformatics.html#workflow-eda-for-transcriptomics"><i class="fa fa-check"></i><b>E.18.1</b> Workflow: EDA for Transcriptomics</a></li>
</ul></li>
<li class="chapter" data-level="E.19" data-path="bioinformatics.html"><a href="bioinformatics.html#dimensionality-reduction"><i class="fa fa-check"></i><b>E.19</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="E.20" data-path="bioinformatics.html"><a href="bioinformatics.html#data-science-skills"><i class="fa fa-check"></i><b>E.20</b> Data Science Skills</a></li>
<li class="chapter" data-level="E.21" data-path="bioinformatics.html"><a href="bioinformatics.html#cross-normalization"><i class="fa fa-check"></i><b>E.21</b> Cross-Normalization</a></li>
<li class="chapter" data-level="E.22" data-path="bioinformatics.html"><a href="bioinformatics.html#clustering"><i class="fa fa-check"></i><b>E.22</b> Clustering</a>
<ul>
<li class="chapter" data-level="E.22.1" data-path="bioinformatics.html"><a href="bioinformatics.html#clustering-evaluation"><i class="fa fa-check"></i><b>E.22.1</b> Clustering Evaluation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="F" data-path="systems-biology-1.html"><a href="systems-biology-1.html"><i class="fa fa-check"></i><b>F</b> Systems Biology</a>
<ul>
<li class="chapter" data-level="F.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-thinking"><i class="fa fa-check"></i><b>F.1</b> Systems Thinking</a>
<ul>
<li class="chapter" data-level="F.1.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#models"><i class="fa fa-check"></i><b>F.1.1</b> MODELS</a></li>
<li class="chapter" data-level="F.1.2" data-path="systems-biology-1.html"><a href="systems-biology-1.html#general-systems-theory"><i class="fa fa-check"></i><b>F.1.2</b> General Systems Theory</a></li>
</ul></li>
<li class="chapter" data-level="F.2" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-biology-2"><i class="fa fa-check"></i><b>F.2</b> Systems Biology</a>
<ul>
<li class="chapter" data-level="F.2.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#hematopoiesis-and-its-disorders_a-systems-biology-approach"><i class="fa fa-check"></i><b>F.2.1</b> Hematopoiesis and its disorders_a systems biology approach</a></li>
</ul></li>
<li class="chapter" data-level="F.3" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-vaccinology-1"><i class="fa fa-check"></i><b>F.3</b> Systems Vaccinology</a>
<ul>
<li class="chapter" data-level="F.3.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-vaccinology---pulendran-nakaya---2010"><i class="fa fa-check"></i><b>F.3.1</b> Systems Vaccinology - Pulendran, Nakaya - 2010</a></li>
</ul></li>
<li class="chapter" data-level="F.4" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-immunology"><i class="fa fa-check"></i><b>F.4</b> Systems Immunology</a></li>
<li class="chapter" data-level="F.5" data-path="systems-biology-1.html"><a href="systems-biology-1.html#artificial-biology"><i class="fa fa-check"></i><b>F.5</b> Artificial Biology</a>
<ul>
<li class="chapter" data-level="F.5.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#principles-of-genetic-circuit-design"><i class="fa fa-check"></i><b>F.5.1</b> Principles of Genetic Circuit Design</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="G" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>G</b> Resources</a>
<ul>
<li class="chapter" data-level="G.1" data-path="resources.html"><a href="resources.html#the-human-cell-atlas"><i class="fa fa-check"></i><b>G.1</b> The Human Cell Atlas</a></li>
<li class="chapter" data-level="G.2" data-path="resources.html"><a href="resources.html#reproducible-data-analysis"><i class="fa fa-check"></i><b>G.2</b> reproducible data analysis</a></li>
<li class="chapter" data-level="G.3" data-path="resources.html"><a href="resources.html#databases"><i class="fa fa-check"></i><b>G.3</b> Databases</a></li>
</ul></li>
<li class="chapter" data-level="H" data-path="statistics-probability.html"><a href="statistics-probability.html"><i class="fa fa-check"></i><b>H</b> Statistics &amp; Probability</a>
<ul>
<li class="chapter" data-level="H.1" data-path="statistics-probability.html"><a href="statistics-probability.html#statistics"><i class="fa fa-check"></i><b>H.1</b> Statistics</a></li>
<li class="chapter" data-level="H.2" data-path="statistics-probability.html"><a href="statistics-probability.html#probability"><i class="fa fa-check"></i><b>H.2</b> Probability</a></li>
</ul></li>
<li class="chapter" data-level="I" data-path="virology.html"><a href="virology.html"><i class="fa fa-check"></i><b>I</b> Virology</a></li>
<li class="chapter" data-level="J" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html"><i class="fa fa-check"></i><b>J</b> Goals and Questions (Original)</a>
<ul>
<li class="chapter" data-level="J.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#goals"><i class="fa fa-check"></i><b>J.1</b> Goals</a>
<ul>
<li class="chapter" data-level="J.1.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#main-goal"><i class="fa fa-check"></i><b>J.1.1</b> Main Goal</a></li>
<li class="chapter" data-level="J.1.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#specific-goals"><i class="fa fa-check"></i><b>J.1.2</b> Specific Goals</a></li>
</ul></li>
<li class="chapter" data-level="J.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#research-questions-1"><i class="fa fa-check"></i><b>J.2</b> Research Questions</a>
<ul>
<li class="chapter" data-level="J.2.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-inflammation-in-the-nose-change-with-sex"><i class="fa fa-check"></i><b>J.2.1</b> Does inflammation in the nose change with sex?</a></li>
<li class="chapter" data-level="J.2.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-inflammation-in-the-nose-change-with-age"><i class="fa fa-check"></i><b>J.2.2</b> Does inflammation in the nose change with age?</a></li>
<li class="chapter" data-level="J.2.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#can-baseline-values-of-inflammation-in-the-nose-predict-who-will-become-colonized-in-the-future"><i class="fa fa-check"></i><b>J.2.3</b> Can baseline values of inflammation in the nose predict who will become colonized in the future?</a></li>
<li class="chapter" data-level="J.2.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#do-any-of-the-cytokines-of-immune-cell-readouts-from-flow-data-correlate-with-spn-density"><i class="fa fa-check"></i><b>J.2.4</b> Do any of the cytokines of immune cell readouts (from Flow data) correlate with Spn density?</a></li>
<li class="chapter" data-level="J.2.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#do-any-of-the-cytokines-correlate-with-flow-cytometry-data"><i class="fa fa-check"></i><b>J.2.5</b> Do any of the cytokines correlate with flow cytometry data?</a></li>
<li class="chapter" data-level="J.2.6" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-the-induction-of-cytokines-differ-after-spn-inoculation-with-sex"><i class="fa fa-check"></i><b>J.2.6</b> Does the induction of cytokines differ after Spn inoculation with sex?</a></li>
<li class="chapter" data-level="J.2.7" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-the-induction-of-cytokines-differ-after-spn-inoculation-with-age"><i class="fa fa-check"></i><b>J.2.7</b> Does the induction of cytokines differ after Spn inoculation with age?</a></li>
<li class="chapter" data-level="J.2.8" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#is-the-response-different-in-natural-carriers"><i class="fa fa-check"></i><b>J.2.8</b> Is the response different in natural carriers?</a></li>
</ul></li>
<li class="chapter" data-level="J.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#specific-questions"><i class="fa fa-check"></i><b>J.3</b> Specific Questions</a>
<ul>
<li class="chapter" data-level="J.3.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#pneumococcal-carriage-load-spn-density"><i class="fa fa-check"></i><b>J.3.1</b> Pneumococcal Carriage Load (<em>Spn Density</em>)</a></li>
<li class="chapter" data-level="J.3.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#gene-expression-rna-seq"><i class="fa fa-check"></i><b>J.3.2</b> Gene Expression (<em>RNA-Seq</em>)</a></li>
<li class="chapter" data-level="J.3.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#cytokines-luminex"><i class="fa fa-check"></i><b>J.3.3</b> Cytokines (<em>Luminex</em>)</a></li>
<li class="chapter" data-level="J.3.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#hormone-levels"><i class="fa fa-check"></i><b>J.3.4</b> Hormone Levels</a></li>
<li class="chapter" data-level="J.3.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#cell-recruitment-flow-cytometry"><i class="fa fa-check"></i><b>J.3.5</b> Cell Recruitment (<em>Flow Cytometry</em>)</a></li>
</ul></li>
<li class="chapter" data-level="J.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#technical-questions"><i class="fa fa-check"></i><b>J.4</b> Technical Questions</a></li>
<li class="chapter" data-level="J.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#sex-effect-on-spn-carriage"><i class="fa fa-check"></i><b>J.5</b> Sex Effect on Spn Carriage</a></li>
<li class="chapter" data-level="J.6" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#ageing-effect-on-spn-carriage"><i class="fa fa-check"></i><b>J.6</b> Ageing Effect on Spn Carriage</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Ageing and Sex Differences in Human Pneumococcal Carriage</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="definitions" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Definitions</h1>
<dl>
<dt>cross-over design</dt>
<dd><p>is a repeated measurements design such that each experimental unit (patient) receives different treatments during the different time periods, i.e., the patients cross over from one treatment to another during the course of the trial</p>
</dd>
<dt>w. r. t</dt>
<dd><p>with respect to, with regard to</p>
</dd>
<dt>Dependent/outcome variable</dt>
<dd><p>A variable whose value depends upon independent variable, is what is being measured in an experiment or evaluated in a mathematical equation. In the equation a = b/c, the dependent variable, a, is determined by the values of b and c.</p>
</dd>
<dt>Response variables</dt>
<dd><p>The response variable is the variable you are measuring and trying to explain. When you have a response variable, it is always paired with one or more explanatory variables. The explanatory variable(s) drives change in the response variable.</p>
</dd>
<dt>Latent variables</dt>
<dd><p>As opposed to observable variables, are variables that are not directly observed but are rather inferred (through a mathematical model) from other variables that are observed (directly measured).</p>
</dd>
<dt>Covariance</dt>
<dd><p>is a measure of the joint variability of two random variables. Covariance provides a measure of the strength of the correlation between two or more sets of random variates.</p>
<p><span class="math inline">\(Cov(x,y) = \Sigma_{i=1,…,N} (x_i - \mu_x)(y_i - \mu_y)/N\)</span></p>
</dd>
<dt>Linear combination</dt>
<dd><p>A linear combination is an expression constructed from a set of terms by multiplying each term by a constant and adding the results (e.g. a linear combination of x and y would be any expression of the form ax + by, where a and b are constants).</p>
</dd>
<dt>Independent Component Analysis (ICA)</dt>
<dd><p>Is a statistical and computational technique for revealing hidden factors that underlie sets of random variables, measurements, or signals. In the model, the data variables are assumed to be linear or nonlinear mixtures of some unknown latent variables, and the mixing system is also unknown. The latent variables are assumed non-gaussian and mutually independent, and they are called the independent components of the observed data. These independent components, also called sources or factors, can be found by ICA.</p>
</dd>
<dt>bipartite graphs</dt>
<dd><p>also called a bigraph, is a set of graph vertices decomposed into two disjoint sets such that no two graph vertices within the same set are adjacent.</p>
</dd>
<dt>Discriminant Analysis</dt>
<dd><p>i.g., the prediction of group membership from the levels of continuous predictor variables</p>
</dd>
<dt>Partial Least Squares (PLS) regression</dt>
<dd><p>Partial least squares (PLS) regression is a technique that reduces the predictors to a smaller set of uncorrelated components and performs least squares regression on these components, instead of on the original data. PLS regression is especially useful when your predictors are highly collinear, or when you have more predictors than observations and ordinary least-squares regression either produces coefficients with high standard errors or fails completely. PLS does not assume that the predictors are fixed, unlike multiple regression. This means that the predictors can be measured with error, making PLS more robust to measurement uncertainty.</p>
</dd>
<dt>Multicollinearity (collinearity)</dt>
<dd><p>is a phenomenon in which one predictor variable in a multiple regression model can be linearly predicted from the others with a substantial degree of accuracy. In this situation the coefficient estimates of the multiple regression may change erratically in response to small changes in the model or the data. Multicollinearity does not reduce the predictive power or reliability of the model as a whole, at least within the sample data set; it only affects calculations regarding individual predictors. That is, a multivariate regression model with collinear predictors can indicate how well the entire bundle of predictors predicts the outcome variable, but it may not give valid results about any individual predictor, or about which predictors are redundant with respect to others.</p>
<p>Multicollinearity occurs when your model includes multiple factors that are correlated not just to your response variable, but also to each other. In other words, it results when you have factors that are a bit redundant.</p>
<p>Multicollinearity increases the standard errors of the coefficients. Increased standard errors in turn means that coefficients for some independent variables may be found not to be significantly different from 0. In other words, by overinflating the standard errors, multicollinearity makes some variables statistically insignificant when they should be significant. Without multicollinearity (and thus, with lower standard errors), those coefficients might be significant.</p>
<p>Severe multicollinearity is a major problem, because it increases the variance of the regression coefficients, making them unstable. The more variance they have, the more difficult it is to interpret the coefficients.</p>
<p>Warning Signs of Multicollinearity :</p>
<ul>
<li><p>A regression coefficient is not significant even though, theoretically, that variable should be highly correlated with Y.</p></li>
<li><p>When you add or delete an X variable, the regression coefficients change dramatically.</p></li>
<li><p>You see a negative regression coefficient when your response should increase along with X.</p></li>
<li><p>You see a positive regression coefficient when the response should decrease as X increases.</p></li>
<li><p>Your X variables have high pairwise correlations.</p></li>
</ul>
<p>Ways to measure multicollinearity: Variance Inflation Factor (VIF): assesses how much the variance of an estimated regression coefficient increases if your predictors are correlated:</p>
<ul>
<li><p>VIF is equal to 1, there is no multicollinearity among factors</p></li>
<li><p>VIF is greater than 1, the predictors may be moderately correlated</p></li>
<li><p>VIF between 5 and 10, indicates high correlation that may be problematic</p></li>
<li><p>VIF above 10, you can assume that the regression coefficients are poorly estimated due to multicollinearity.</p></li>
</ul>
<p>Dealing with Multicolinearity (VIF near or above 5):</p>
<p>Remove highly correlated predictors from the model. If you have two or more factors with a high VIF, remove one from the model. Because they supply redundant information, removing one of the correlated factors usually doesn’t drastically reduce the R-squared. Consider using stepwise regression, best subsets regression, or specialized knowledge of the data set to remove these variables. Select the model that has the highest R-squared value.</p>
<p>Use Partial Least Squares Regression (PLS) or Principal Components Analysis, regression methods that cut the number of predictors to a smaller set of uncorrelated components.</p>
</dd>
<dt>Multivariate/Multivariable Analysis</dt>
<dd><p>(linear) observation and analysis of more than one statistical outcome variable at a time.</p>
<p>Multivariate is used for the analysis with multiple outcomes/dependent variables, whereas multivariable is used for the analysis with multiple explanatory/independent variables. However, they are sometimes used interchangeably in the literature as not many researchers are aware of the difference.</p>
<p>A simple linear regression model has a continuous outcome and one predictor, whereas a multiple or multivariable linear regression model has a continuous outcome and multiple predictors (continuous or categorical). A simple linear regression model would have the form y = a + xb + e. By contrast, a multivariable or multiple linear regression model would take the form y = a + x1b1 + x2b2 + … + xkbk + e, where y is a continuous dependent variable, x is a single predictor in the simple regression model, and x1, x2, …, xk are the predictors in the multivariable model.</p>
<p>Multivariate, by contrast, refers to the modeling of data that are often derived from longitudinal studies, wherein an outcome is measured for the same individual at multiple time points (repeated measures), or the modeling of nested/clustered data, wherein there are multiple individuals in each cluster. A multivariate linear regression model would have the form</p>
</dd>
<dt>Orthogonal transformation</dt>
<dd><p>An orthogonal transformation is a linear transformation T:V-&gt;V which preserves a symmetric inner product: preserves lengths of vectors and angles between vectors</p>
</dd>
<dt>Principal Component Analysis</dt>
<dd><p>Linear transformations of independent varibles of high-dimentional data into lower-dimentionality space, seeking the best directions in the data that account for most of the variability</p>
<pre><code>Uses: Mainly used for dimensionality reduction and feature extraction   in large datasets.

Others: PCA transforms and project high-dimensional data in a low-dimensional space, by lineary combining the original independent variables in new orthogonal and independent variables that can best explain/highlight the variance present in the data.</code></pre>
</dd>
<dt>principal components</dt>
<dd><p>artificial variables that are linear combinations of the original variables</p>
</dd>
<dt>ill-posed problem</dt>
<dd><p>According to Jacques Hadamard, mathematical models of physical phenomena should have the properties that:</p>
<ol style="list-style-type: decimal">
<li>a solution exists</li>
<li>the solution is unique</li>
<li>the solution’s behavior changes continuously with the initial conditions</li>
</ol>
<p>If the problem is well-posed, then it stands a good chance of solution on a computer using a stable algorithm. If it is not well-posed, it needs to be re-formulated for numerical treatment. Typically this involves including additional assumptions, such as smoothness of solution. This process is known as regularization.</p>
<p>Even if a problem is well-posed, it may still be ill-conditioned, meaning that a small error in the initial data can result in much larger errors in the answers. Problems in nonlinear complex systems (so called chaotic systems) provide well-known examples of instability. An ill-conditioned problem is indicated by a large condition number.</p>
</dd>
<dt>regularization</dt>
<dd><p>is the process of adding information in order to solve an ill-posed problem or to prevent overfitting</p>
</dd>
<dt>overfitting</dt>
<dd><p>is the production of an analysis that corresponds too closely or exactly to a particular set of data, and may therefore fail to fit additiona data or predict future observations reliably. An overfitted model is a statistical lmodel that contains more paramenters than can be justified by the data. The essence of overfitting is to have unknowingly extracted some of the residual variation (i.e., the noise) as if that variation represented underlying model structure.</p>
</dd>
<dt>underfitting</dt>
<dd><p>occurs when a statistical model cannot adequately capture the underlying structure of the data. An underfitted model is a model where some parameters or terms that would appear in a correctly specified model are missing. Underfitting would occur, e.g., when fitting a linear model to non-linear data, tending to have poor predictive performance.</p>
</dd>
<dt>LASSO</dt>
<dd><p><em>Least Absolute Shrinkage and Selection Operator.</em> Regression analysis method that performs both variable selection and regularization in o\`\`rder to enhance the prediction accuracy and interpretability of the statistical model it produces. Lasso was introduced in order to improve the prediction accuracy and interpretability of regression models by altering the model fitting process to select only a subset of the provided covariates for use in the final model rather than using all of them</p>
<p>Is a regression method that involves penalizing the absolute size of the regression coefficients. By penalizing (or equivalently constraining the sum of the absolute values of the estimates) you end up in a situation where some of the parameter estimates may be exactly zero. The larger the penalty applied, the further estimates are shrunk towards zero. This is convenient when we want some automatic feature/variable selection, or when dealing with highly correlated predictors, where standard regression will usually have regression coefficients that are ‘too large.’</p>
<p>Generalizations of lasso:</p>
<ul>
<li><p>Elastic net</p></li>
<li><p>Group lasso</p></li>
<li><p>Fused lasso</p></li>
<li><p>Quasi-norms and bridge regression</p></li>
<li><p>Adaptive lasso</p></li>
</ul>
</dd>
<dt>Factor Analysis</dt>
<dd><p><a href="https://www.theanalysisfactor.com/factor-analysis-1-introduction/">Factor Analysis</a> is a statistical method used to describe variability among observed, correlated variables in terms of a potentially lower number of unobserved variables called factors. Is a useful tool for investigating variable relationships for complex concepts such as socioeconomic status, dietary patterns, or psychological scales. Allows researchers to investigate concepts that are not easily measured directly by collapsing a large number of variables into a few interpretable underlying factors.</p>
<p>The key concept of factor analysis is that multiple observed variables have similar patterns of responses because they are all associated with a latent (i.e. not directly measured) variable. In every factor analysis, there are the same number of factors as there are variables. Each factor captures a certain amount of the overall variance in the observed variables, and the factors are always listed in order of how much variation they explain.</p>
<p>The eigenvalue is a measure of how much of the variance of the observed variables a factor explains. Any factor with an eigenvalue ≥1 explains more variance than a single observed variable. The relationship of each variable to the underlying factor is expressed by the so-called factor loading. Since factor loadings can be interpreted like standardized regression coefficients, one could also say that the variable income has a correlation of 0.65 with Factor 1. This would be considered a strong association for a factor analysis in most research fields.</p>
</dd>
</dl>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["phd_thesis.pdf", "phd_thesis.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
