<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Background Knowledge | Ageing and Sex Differences in Human Pneumococcal Carriage</title>
  <meta name="description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Background Knowledge | Ageing and Sex Differences in Human Pneumococcal Carriage" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Background Knowledge | Ageing and Sex Differences in Human Pneumococcal Carriage" />
  
  <meta name="twitter:description" content="Computational Systems Biology for pneumonia vaccine development using human challenge models" />
  

<meta name="author" content="Fernando Marcon Passos" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="research-questions.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Sex and Ageing Effects on Pneumococcal Carriage</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>_</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Background Knowledge</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#pneumonia-and-the-pneumococcal-diseases"><i class="fa fa-check"></i><b>1.1</b> Pneumonia and the Pneumococcal diseases</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#streptococcus-pneumoniae"><i class="fa fa-check"></i><b>1.2</b> Streptococcus pneumoniae</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#human-protection-to-pneumococcal-diseases"><i class="fa fa-check"></i><b>1.3</b> Human Protection to Pneumococcal Diseases</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#the-effects-of-sex-and-ageing-on-human-immunity"><i class="fa fa-check"></i><b>1.4</b> The Effects of Sex and Ageing on Human Immunity</a></li>
<li class="chapter" data-level="1.5" data-path="intro.html"><a href="intro.html#the-experimental-human-pneumococcal-carriage-model"><i class="fa fa-check"></i><b>1.5</b> The Experimental Human Pneumococcal Carriage Model</a></li>
<li class="chapter" data-level="1.6" data-path="intro.html"><a href="intro.html#systems-vaccinology"><i class="fa fa-check"></i><b>1.6</b> Systems Vaccinology</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="research-questions.html"><a href="research-questions.html"><i class="fa fa-check"></i><b>2</b> Research Questions</a></li>
<li class="chapter" data-level="3" data-path="project-plan.html"><a href="project-plan.html"><i class="fa fa-check"></i><b>3</b> Project Plan</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a>
<ul>
<li class="chapter" data-level="4.1" data-path="methods.html"><a href="methods.html#overall-experimental-design-data-and-analysis"><i class="fa fa-check"></i><b>4.1</b> Overall Experimental Design, Data and Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="methods.html"><a href="methods.html#experimental-designs-and-data-measurements"><i class="fa fa-check"></i><b>4.1.1</b> Experimental Designs and Data Measurements</a></li>
<li class="chapter" data-level="4.1.2" data-path="methods.html"><a href="methods.html#overall-data-types-and-preprocessing"><i class="fa fa-check"></i><b>4.1.2</b> Overall Data Types and Preprocessing</a></li>
<li class="chapter" data-level="4.1.3" data-path="methods.html"><a href="methods.html#biological-pathways"><i class="fa fa-check"></i><b>4.1.3</b> Biological Pathways</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="methods.html"><a href="methods.html#proposed-solution"><i class="fa fa-check"></i><b>4.2</b> Proposed Solution</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="methods.html"><a href="methods.html#overall-data-analysis"><i class="fa fa-check"></i><b>4.2.1</b> Overall Data Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="methods.html"><a href="methods.html#transcriptomics-analysis-pipeline"><i class="fa fa-check"></i><b>4.3</b> Transcriptomics Analysis Pipeline</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="methods.html"><a href="methods.html#quality-control"><i class="fa fa-check"></i><b>4.3.1</b> Quality Control</a></li>
<li class="chapter" data-level="4.3.2" data-path="methods.html"><a href="methods.html#biomarkers"><i class="fa fa-check"></i><b>4.3.2</b> Biomarkers</a></li>
<li class="chapter" data-level="4.3.3" data-path="methods.html"><a href="methods.html#pathway-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Pathway Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="methods.html"><a href="methods.html#pipeline"><i class="fa fa-check"></i><b>4.3.4</b> Pipeline</a></li>
<li class="chapter" data-level="4.3.5" data-path="methods.html"><a href="methods.html#section-1"><i class="fa fa-check"></i><b>4.3.5</b> </a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="methods.html"><a href="methods.html#network-analysis"><i class="fa fa-check"></i><b>4.4</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="methods.html"><a href="methods.html#co-expression-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Co-expression Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><a href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><i class="fa fa-check"></i><b>5</b> Data Driven Analysis of Susceptibility to Pneumococcal Carriage</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html"><a href="data-driven-analysis-of-susceptibility-to-pneumococcal-carriage.html#q1-is-developing-carriage-random"><i class="fa fa-check"></i><b>5.1</b> Q1: Is developing carriage random?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="definitions.html"><a href="definitions.html"><i class="fa fa-check"></i><b>6</b> Definitions</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="essential-literature.html"><a href="essential-literature.html"><i class="fa fa-check"></i><b>A</b> Essential Literature</a>
<ul>
<li class="chapter" data-level="A.1" data-path="essential-literature.html"><a href="essential-literature.html#thesis"><i class="fa fa-check"></i><b>A.1</b> Thesis</a></li>
<li class="chapter" data-level="A.2" data-path="essential-literature.html"><a href="essential-literature.html#articles"><i class="fa fa-check"></i><b>A.2</b> Articles</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="essential-literature.html"><a href="essential-literature.html#pneumococcal-diseases"><i class="fa fa-check"></i><b>A.2.1</b> Pneumococcal Diseases</a></li>
<li class="chapter" data-level="A.2.2" data-path="essential-literature.html"><a href="essential-literature.html#sex-differences"><i class="fa fa-check"></i><b>A.2.2</b> Sex Differences</a></li>
<li class="chapter" data-level="A.2.3" data-path="essential-literature.html"><a href="essential-literature.html#ageing"><i class="fa fa-check"></i><b>A.2.3</b> Ageing</a></li>
<li class="chapter" data-level="A.2.4" data-path="essential-literature.html"><a href="essential-literature.html#systems-biology"><i class="fa fa-check"></i><b>A.2.4</b> Systems Biology</a></li>
<li class="chapter" data-level="A.2.5" data-path="essential-literature.html"><a href="essential-literature.html#bioinformatics-and-methodologies"><i class="fa fa-check"></i><b>A.2.5</b> Bioinformatics and Methodologies</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html"><i class="fa fa-check"></i><b>B</b> Pneumococcal Diseases and Immunity</a>
<ul>
<li class="chapter" data-level="B.1" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html#controlled-human-infection-and-rechallenge-with-streptococcus-pneumoniae-reveals-the-protecticve-efficacy-of-carriage-in-healty-adults"><i class="fa fa-check"></i><b>B.1</b> Controlled Human Infection and Rechallenge with Streptococcus pneumoniae Reveals the Protecticve Efficacy of Carriage in Healty Adults</a></li>
<li class="chapter" data-level="B.2" data-path="pneumococcal-diseases-and-immunity.html"><a href="pneumococcal-diseases-and-immunity.html#inflammation-induced-by-influenza-virus-impairs-innate-control-of-human-pneumococcal-carriage"><i class="fa fa-check"></i><b>B.2</b> Inflammation induced by influenza virus impairs innate control of human pneumococcal carriage</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="sex-differences-1.html"><a href="sex-differences-1.html"><i class="fa fa-check"></i><b>C</b> Sex Differences</a></li>
<li class="chapter" data-level="D" data-path="ageing-1.html"><a href="ageing-1.html"><i class="fa fa-check"></i><b>D</b> Ageing</a>
<ul>
<li class="chapter" data-level="D.1" data-path="ageing-1.html"><a href="ageing-1.html#the-hallmarks-of-aging"><i class="fa fa-check"></i><b>D.1</b> The Hallmarks of Aging</a></li>
<li class="chapter" data-level="D.2" data-path="ageing-1.html"><a href="ageing-1.html#therapies"><i class="fa fa-check"></i><b>D.2</b> Therapies</a></li>
<li class="chapter" data-level="D.3" data-path="ageing-1.html"><a href="ageing-1.html#senescent-cells"><i class="fa fa-check"></i><b>D.3</b> Senescent Cells</a></li>
<li class="chapter" data-level="D.4" data-path="ageing-1.html"><a href="ageing-1.html#the-role-of-senescent-cells-in-aging"><i class="fa fa-check"></i><b>D.4</b> The Role of Senescent Cells in Aging</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="bioinformatics.html"><a href="bioinformatics.html"><i class="fa fa-check"></i><b>E</b> Bioinformatics</a>
<ul>
<li class="chapter" data-level="E.1" data-path="bioinformatics.html"><a href="bioinformatics.html#bioinformatics-essentials"><i class="fa fa-check"></i><b>E.1</b> Bioinformatics Essentials:</a></li>
<li class="chapter" data-level="E.2" data-path="bioinformatics.html"><a href="bioinformatics.html#differential-expression-analysis"><i class="fa fa-check"></i><b>E.2</b> Differential Expression Analysis</a>
<ul>
<li class="chapter" data-level="E.2.1" data-path="bioinformatics.html"><a href="bioinformatics.html#empirical-assessment-of-analysis-workflows-for-differential-expression-analysis-of-human-samples-using-rna-seq"><i class="fa fa-check"></i><b>E.2.1</b> Empirical assessment of analysis workflows for differential expression analysis of human samples using RNA-Seq</a></li>
</ul></li>
<li class="chapter" data-level="E.3" data-path="bioinformatics.html"><a href="bioinformatics.html#network-analysis-1"><i class="fa fa-check"></i><b>E.3</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="E.3.1" data-path="bioinformatics.html"><a href="bioinformatics.html#genemania-a-real-time-multiple-association-network-integration-algorithm-for-predicting-gene-function"><i class="fa fa-check"></i><b>E.3.1</b> 2008 - GeneMANIA: a real-time multiple association network integration algorithm for predicting gene function</a></li>
<li class="chapter" data-level="E.3.2" data-path="bioinformatics.html"><a href="bioinformatics.html#the-genemania-prediction-server-biological-network-integration-for-gene-prioritization-and-predicting-gene-function"><i class="fa fa-check"></i><b>E.3.2</b> 2010 - The GeneMANIA prediction server: biological network integration for gene prioritization and predicting gene function</a></li>
</ul></li>
<li class="chapter" data-level="E.4" data-path="bioinformatics.html"><a href="bioinformatics.html#multidimensional-scaling"><i class="fa fa-check"></i><b>E.4</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="E.5" data-path="bioinformatics.html"><a href="bioinformatics.html#next-generation-sequencing"><i class="fa fa-check"></i><b>E.5</b> Next-Generation Sequencing</a>
<ul>
<li class="chapter" data-level="E.5.1" data-path="bioinformatics.html"><a href="bioinformatics.html#rna-seq-analysis"><i class="fa fa-check"></i><b>E.5.1</b> RNA-Seq Analysis</a></li>
<li class="chapter" data-level="E.5.2" data-path="bioinformatics.html"><a href="bioinformatics.html#microbiome-data-analysis"><i class="fa fa-check"></i><b>E.5.2</b> microbiome data analysis</a></li>
</ul></li>
<li class="chapter" data-level="E.6" data-path="bioinformatics.html"><a href="bioinformatics.html#omics"><i class="fa fa-check"></i><b>E.6</b> OMICS</a>
<ul>
<li class="chapter" data-level="E.6.1" data-path="bioinformatics.html"><a href="bioinformatics.html#integromics-an-r-package-to-unravel-relationships-between-two-omics-datasets"><i class="fa fa-check"></i><b>E.6.1</b> integrOmics an R package to unravel relationships between two omics datasets</a></li>
<li class="chapter" data-level="E.6.2" data-path="bioinformatics.html"><a href="bioinformatics.html#visualising-associations-between-paired-omics-data-sets"><i class="fa fa-check"></i><b>E.6.2</b> Visualising associations between paired “omics” data sets</a></li>
</ul></li>
<li class="chapter" data-level="E.7" data-path="bioinformatics.html"><a href="bioinformatics.html#principal-component-analysis"><i class="fa fa-check"></i><b>E.7</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="E.8" data-path="bioinformatics.html"><a href="bioinformatics.html#mixomics"><i class="fa fa-check"></i><b>E.8</b> MixOmics</a>
<ul>
<li class="chapter" data-level="E.8.1" data-path="bioinformatics.html"><a href="bioinformatics.html#diablo"><i class="fa fa-check"></i><b>E.8.1</b> DIABLO</a></li>
<li class="chapter" data-level="E.8.2" data-path="bioinformatics.html"><a href="bioinformatics.html#exploratory-data-analysis-with-mixomics"><i class="fa fa-check"></i><b>E.8.2</b> EXPLORATORY DATA ANALYSIS WITH mixOMICS</a></li>
<li class="chapter" data-level="E.8.3" data-path="bioinformatics.html"><a href="bioinformatics.html#independent-component-analysis-ica"><i class="fa fa-check"></i><b>E.8.3</b> - Independent Component Analysis (ICA)</a></li>
<li class="chapter" data-level="E.8.4" data-path="bioinformatics.html"><a href="bioinformatics.html#independent-principal-component-analysis-ipca"><i class="fa fa-check"></i><b>E.8.4</b> Independent Principal Component Analysis (IPCA)</a></li>
<li class="chapter" data-level="E.8.5" data-path="bioinformatics.html"><a href="bioinformatics.html#mixomics---an-r-package-for-omics-feature-selection-and-multiple-data-integration_2017"><i class="fa fa-check"></i><b>E.8.5</b> mixOmics - An R package for omics feature selection and multiple data integration_2017</a></li>
<li class="chapter" data-level="E.8.6" data-path="bioinformatics.html"><a href="bioinformatics.html#partial-least-squares-regression-pls-regression"><i class="fa fa-check"></i><b>E.8.6</b> Partial Least Squares regression (PLS regression)</a></li>
<li class="chapter" data-level="E.8.7" data-path="bioinformatics.html"><a href="bioinformatics.html#variable-selection-for-generalized-canonical-correlation-analysis"><i class="fa fa-check"></i><b>E.8.7</b> Variable selection for generalized canonical correlation analysis</a></li>
<li class="chapter" data-level="E.8.8" data-path="bioinformatics.html"><a href="bioinformatics.html#unravelling-omics-data-with-the-mixomics-r-package-ppt"><i class="fa fa-check"></i><b>E.8.8</b> Unravelling “omics” data with the mixOmics R package (ppt)</a></li>
</ul></li>
<li class="chapter" data-level="E.9" data-path="bioinformatics.html"><a href="bioinformatics.html#normalization"><i class="fa fa-check"></i><b>E.9</b> Normalization</a>
<ul>
<li class="chapter" data-level="E.9.1" data-path="bioinformatics.html"><a href="bioinformatics.html#the-universal-expression-code-upc-algorithm-consists-of-two-main-steps"><i class="fa fa-check"></i><b>E.9.1</b> The Universal exPression Code (UPC) algorithm consists of two main steps:</a></li>
</ul></li>
<li class="chapter" data-level="E.10" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-analysis"><i class="fa fa-check"></i><b>E.10</b> Time-Series Analysis</a>
<ul>
<li class="chapter" data-level="E.10.1" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-clustering"><i class="fa fa-check"></i><b>E.10.1</b> Time-Series Clustering</a></li>
<li class="chapter" data-level="E.10.2" data-path="bioinformatics.html"><a href="bioinformatics.html#time-series-options"><i class="fa fa-check"></i><b>E.10.2</b> Time-Series Options</a></li>
</ul></li>
<li class="chapter" data-level="E.11" data-path="bioinformatics.html"><a href="bioinformatics.html#survival-analysis"><i class="fa fa-check"></i><b>E.11</b> Survival analysis</a></li>
<li class="chapter" data-level="E.12" data-path="bioinformatics.html"><a href="bioinformatics.html#data-integration"><i class="fa fa-check"></i><b>E.12</b> Data Integration</a>
<ul>
<li class="chapter" data-level="E.12.1" data-path="bioinformatics.html"><a href="bioinformatics.html#sparse-canonical-methods-for-biological-data-integration-application-to-a-cross-platform-study"><i class="fa fa-check"></i><b>E.12.1</b> Sparse canonical methods for biological data integration: application to a cross-platform study</a></li>
</ul></li>
<li class="chapter" data-level="E.13" data-path="bioinformatics.html"><a href="bioinformatics.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>E.13</b> Single-Cell RNA-Seq</a>
<ul>
<li class="chapter" data-level="E.13.1" data-path="bioinformatics.html"><a href="bioinformatics.html#analysis"><i class="fa fa-check"></i><b>E.13.1</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="E.14" data-path="bioinformatics.html"><a href="bioinformatics.html#reproducible-data-science"><i class="fa fa-check"></i><b>E.14</b> Reproducible Data Science</a></li>
<li class="chapter" data-level="E.15" data-path="bioinformatics.html"><a href="bioinformatics.html#machine-learning-vs-de-analysis"><i class="fa fa-check"></i><b>E.15</b> machine learning vs DE Analysis</a></li>
<li class="chapter" data-level="E.16" data-path="bioinformatics.html"><a href="bioinformatics.html#dealing-with-missing-data"><i class="fa fa-check"></i><b>E.16</b> Dealing with Missing Data</a></li>
<li class="chapter" data-level="E.17" data-path="bioinformatics.html"><a href="bioinformatics.html#feature-selection"><i class="fa fa-check"></i><b>E.17</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="E.17.1" data-path="bioinformatics.html"><a href="bioinformatics.html#features-and-feature-engineering"><i class="fa fa-check"></i><b>E.17.1</b> Features and feature engineering</a></li>
</ul></li>
<li class="chapter" data-level="E.18" data-path="bioinformatics.html"><a href="bioinformatics.html#exploratory-data-anaysis-eda"><i class="fa fa-check"></i><b>E.18</b> Exploratory Data Anaysis (EDA)</a>
<ul>
<li class="chapter" data-level="E.18.1" data-path="bioinformatics.html"><a href="bioinformatics.html#workflow-eda-for-transcriptomics"><i class="fa fa-check"></i><b>E.18.1</b> Workflow: EDA for Transcriptomics</a></li>
</ul></li>
<li class="chapter" data-level="E.19" data-path="bioinformatics.html"><a href="bioinformatics.html#dimensionality-reduction"><i class="fa fa-check"></i><b>E.19</b> Dimensionality Reduction</a></li>
<li class="chapter" data-level="E.20" data-path="bioinformatics.html"><a href="bioinformatics.html#data-science-skills"><i class="fa fa-check"></i><b>E.20</b> Data Science Skills</a></li>
<li class="chapter" data-level="E.21" data-path="bioinformatics.html"><a href="bioinformatics.html#cross-normalization"><i class="fa fa-check"></i><b>E.21</b> Cross-Normalization</a></li>
<li class="chapter" data-level="E.22" data-path="bioinformatics.html"><a href="bioinformatics.html#clustering"><i class="fa fa-check"></i><b>E.22</b> Clustering</a>
<ul>
<li class="chapter" data-level="E.22.1" data-path="bioinformatics.html"><a href="bioinformatics.html#clustering-evaluation"><i class="fa fa-check"></i><b>E.22.1</b> Clustering Evaluation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="F" data-path="systems-biology-1.html"><a href="systems-biology-1.html"><i class="fa fa-check"></i><b>F</b> Systems Biology</a>
<ul>
<li class="chapter" data-level="F.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-thinking"><i class="fa fa-check"></i><b>F.1</b> Systems Thinking</a>
<ul>
<li class="chapter" data-level="F.1.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#models"><i class="fa fa-check"></i><b>F.1.1</b> MODELS</a></li>
<li class="chapter" data-level="F.1.2" data-path="systems-biology-1.html"><a href="systems-biology-1.html#general-systems-theory"><i class="fa fa-check"></i><b>F.1.2</b> General Systems Theory</a></li>
</ul></li>
<li class="chapter" data-level="F.2" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-biology-2"><i class="fa fa-check"></i><b>F.2</b> Systems Biology</a>
<ul>
<li class="chapter" data-level="F.2.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#hematopoiesis-and-its-disorders_a-systems-biology-approach"><i class="fa fa-check"></i><b>F.2.1</b> Hematopoiesis and its disorders_a systems biology approach</a></li>
</ul></li>
<li class="chapter" data-level="F.3" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-vaccinology-1"><i class="fa fa-check"></i><b>F.3</b> Systems Vaccinology</a>
<ul>
<li class="chapter" data-level="F.3.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-vaccinology---pulendran-nakaya---2010"><i class="fa fa-check"></i><b>F.3.1</b> Systems Vaccinology - Pulendran, Nakaya - 2010</a></li>
</ul></li>
<li class="chapter" data-level="F.4" data-path="systems-biology-1.html"><a href="systems-biology-1.html#systems-immunology"><i class="fa fa-check"></i><b>F.4</b> Systems Immunology</a></li>
<li class="chapter" data-level="F.5" data-path="systems-biology-1.html"><a href="systems-biology-1.html#artificial-biology"><i class="fa fa-check"></i><b>F.5</b> Artificial Biology</a>
<ul>
<li class="chapter" data-level="F.5.1" data-path="systems-biology-1.html"><a href="systems-biology-1.html#principles-of-genetic-circuit-design"><i class="fa fa-check"></i><b>F.5.1</b> Principles of Genetic Circuit Design</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="G" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>G</b> Resources</a>
<ul>
<li class="chapter" data-level="G.1" data-path="resources.html"><a href="resources.html#the-human-cell-atlas"><i class="fa fa-check"></i><b>G.1</b> The Human Cell Atlas</a></li>
<li class="chapter" data-level="G.2" data-path="resources.html"><a href="resources.html#reproducible-data-analysis"><i class="fa fa-check"></i><b>G.2</b> reproducible data analysis</a></li>
<li class="chapter" data-level="G.3" data-path="resources.html"><a href="resources.html#databases"><i class="fa fa-check"></i><b>G.3</b> Databases</a></li>
</ul></li>
<li class="chapter" data-level="H" data-path="statistics-probability.html"><a href="statistics-probability.html"><i class="fa fa-check"></i><b>H</b> Statistics &amp; Probability</a>
<ul>
<li class="chapter" data-level="H.1" data-path="statistics-probability.html"><a href="statistics-probability.html#statistics"><i class="fa fa-check"></i><b>H.1</b> Statistics</a></li>
<li class="chapter" data-level="H.2" data-path="statistics-probability.html"><a href="statistics-probability.html#probability"><i class="fa fa-check"></i><b>H.2</b> Probability</a></li>
</ul></li>
<li class="chapter" data-level="I" data-path="virology.html"><a href="virology.html"><i class="fa fa-check"></i><b>I</b> Virology</a></li>
<li class="chapter" data-level="J" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html"><i class="fa fa-check"></i><b>J</b> Goals and Questions (Original)</a>
<ul>
<li class="chapter" data-level="J.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#goals"><i class="fa fa-check"></i><b>J.1</b> Goals</a>
<ul>
<li class="chapter" data-level="J.1.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#main-goal"><i class="fa fa-check"></i><b>J.1.1</b> Main Goal</a></li>
<li class="chapter" data-level="J.1.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#specific-goals"><i class="fa fa-check"></i><b>J.1.2</b> Specific Goals</a></li>
</ul></li>
<li class="chapter" data-level="J.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#research-questions-1"><i class="fa fa-check"></i><b>J.2</b> Research Questions</a>
<ul>
<li class="chapter" data-level="J.2.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-inflammation-in-the-nose-change-with-sex"><i class="fa fa-check"></i><b>J.2.1</b> Does inflammation in the nose change with sex?</a></li>
<li class="chapter" data-level="J.2.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-inflammation-in-the-nose-change-with-age"><i class="fa fa-check"></i><b>J.2.2</b> Does inflammation in the nose change with age?</a></li>
<li class="chapter" data-level="J.2.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#can-baseline-values-of-inflammation-in-the-nose-predict-who-will-become-colonized-in-the-future"><i class="fa fa-check"></i><b>J.2.3</b> Can baseline values of inflammation in the nose predict who will become colonized in the future?</a></li>
<li class="chapter" data-level="J.2.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#do-any-of-the-cytokines-of-immune-cell-readouts-from-flow-data-correlate-with-spn-density"><i class="fa fa-check"></i><b>J.2.4</b> Do any of the cytokines of immune cell readouts (from Flow data) correlate with Spn density?</a></li>
<li class="chapter" data-level="J.2.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#do-any-of-the-cytokines-correlate-with-flow-cytometry-data"><i class="fa fa-check"></i><b>J.2.5</b> Do any of the cytokines correlate with flow cytometry data?</a></li>
<li class="chapter" data-level="J.2.6" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-the-induction-of-cytokines-differ-after-spn-inoculation-with-sex"><i class="fa fa-check"></i><b>J.2.6</b> Does the induction of cytokines differ after Spn inoculation with sex?</a></li>
<li class="chapter" data-level="J.2.7" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#does-the-induction-of-cytokines-differ-after-spn-inoculation-with-age"><i class="fa fa-check"></i><b>J.2.7</b> Does the induction of cytokines differ after Spn inoculation with age?</a></li>
<li class="chapter" data-level="J.2.8" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#is-the-response-different-in-natural-carriers"><i class="fa fa-check"></i><b>J.2.8</b> Is the response different in natural carriers?</a></li>
</ul></li>
<li class="chapter" data-level="J.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#specific-questions"><i class="fa fa-check"></i><b>J.3</b> Specific Questions</a>
<ul>
<li class="chapter" data-level="J.3.1" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#pneumococcal-carriage-load-spn-density"><i class="fa fa-check"></i><b>J.3.1</b> Pneumococcal Carriage Load (<em>Spn Density</em>)</a></li>
<li class="chapter" data-level="J.3.2" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#gene-expression-rna-seq"><i class="fa fa-check"></i><b>J.3.2</b> Gene Expression (<em>RNA-Seq</em>)</a></li>
<li class="chapter" data-level="J.3.3" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#cytokines-luminex"><i class="fa fa-check"></i><b>J.3.3</b> Cytokines (<em>Luminex</em>)</a></li>
<li class="chapter" data-level="J.3.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#hormone-levels"><i class="fa fa-check"></i><b>J.3.4</b> Hormone Levels</a></li>
<li class="chapter" data-level="J.3.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#cell-recruitment-flow-cytometry"><i class="fa fa-check"></i><b>J.3.5</b> Cell Recruitment (<em>Flow Cytometry</em>)</a></li>
</ul></li>
<li class="chapter" data-level="J.4" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#technical-questions"><i class="fa fa-check"></i><b>J.4</b> Technical Questions</a></li>
<li class="chapter" data-level="J.5" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#sex-effect-on-spn-carriage"><i class="fa fa-check"></i><b>J.5</b> Sex Effect on Spn Carriage</a></li>
<li class="chapter" data-level="J.6" data-path="goals-and-questions-original.html"><a href="goals-and-questions-original.html#ageing-effect-on-spn-carriage"><i class="fa fa-check"></i><b>J.6</b> Ageing Effect on Spn Carriage</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Ageing and Sex Differences in Human Pneumococcal Carriage</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> Background Knowledge</h1>
<div id="pneumonia-and-the-pneumococcal-diseases" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Pneumonia and the Pneumococcal diseases</h2>
<p>Pneumococcal diseases (PD) constitutes several infectious diseases sharing the same causal factor: <em>Streptococcus pneumoniae</em> (Spn). Examples of such illnesses can range from harmless infections in the middle ear (<em>otitis</em>) and sinus (<em>sinusitis</em>) to more serious and life-threatening pneumonia, meningitis and sepsis. Despite most of such infections be quite common and mild, complications may occur, increasing the severity, originating serious health problems and impairing the chances of survival. Such a scenario is more likely to occur to the parcels of the population with higher exposure to risk factors, vulnerabilities or compromised overall health. Indeed, 90% of deaths caused by pneumonia occurs in countries with low- to middle-income levels, the worst scenario situated in African countries. This might be associated with the high number of people immunocompromised due to HIV infection.</p>
<p>Pneumonia is one of the most common infectious diseases worldwide, known for its significant mortality and frequent need for intensive care support, once respiratory insufficiency and involvement of multiple organs are common complications (Leoni &amp; Rello, 2017) ⁠. Especially among children, elderly and patients with pre-morbid conditions, is associated with a high risk of mortality (Fine et al., 1996) ⁠.</p>
<p>The burden of pneumonia to the human population can be grasped by the fact that it is the leading cause of death in children worldwide, killing more children bellow 5 years than any other diseases, and that its causative agent is enlisted as one of the 12 priority pathogens since 2017 according to the World Health Organization (WHO, 2017)⁠. Furthermore, community-acquired pneumonia (CAP), the most common type of pneumonia, is ranked as the third major cause of death and sepsis in developed countries (Niederman et al., 2001).</p>
</div>
<div id="streptococcus-pneumoniae" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Streptococcus pneumoniae</h2>
<p><em>Streptococcus pneumoniae</em> (Spn) is the leading causes of pneumococcal diseases, being responsible for a variety of related infections. These pathogens are Gram-positive bacteria, coated by a polysaccharide capsule which protect them from being phagocyted. More than 90 chemically and immunologically distinct types were already described (Brugger et al., 2016), several of which can cause several infections types of pneumococcal diseases (O’Brien et al., 2009)⁠.</p>
<p>A significant proportion of the human population is carriers of Spn in their nasal mucosa, with up to 27-65% of children and less 10% of the adults. Even though they are opportunistic pathogens (Orihuela et al., 2009), its transmission is highly dependent on the carriage, which is a commensal type of relationship. The success of transmission requires close contact with a carrier(s), with the chances of success increase if the carriers are very young or during drier and colder seasons due to increased fluid secretions in the airways, and also because it’s more likely to occur in conjunction with viral infections of the upper respiratory tract (URT).</p>
<p>Once transmitted to the airways of a new host by several possible means, they can spread by the throat to both upper and lower respiratory tracts. Depending on its spreading patterns, can be classified into noninvasive and invasive types. The first one colonizes the mucosal superficies of URTs, being less serious because it remains on the outsides of major organs or the blood. Can cause inflammation of the middle ear (otitis media) and sinus (sinusitis) (O’Brien et al., 2009). However, the second and worst type can invade major organs, causing bacteremia, inflammation of the meninges (meningitis) and even life-threatening infection responses by the body (sepsis) and lung diseases (pneumonia). Sometimes might also infect the bones (osteomyelitis) and joints (arthritis).</p>
</div>
<div id="human-protection-to-pneumococcal-diseases" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Human Protection to Pneumococcal Diseases</h2>
<p>The host defense mechanisms against pneumococcus pulmonary infection involve both innate and adaptive immune systems, the first one being rapid but unspecific and the second one being specific but slower.</p>
<p>The first line of defence consists mostly in physical barriers, such as mucus, and also phagocytic and inflammatory responses elicited by the respiratory tract (Wilson et al., 2015). The pathogen is recognized through pattern recognition receptors (PRRs), which can identify important microbial structures like Pathogen Associated Molecular Patterns (PAMPs) and Danger-Associated Molecular Patterns (DAMPs) (Koppe, Suttorp, &amp; Opitz, 2012)⁠. Not only the well-studied Toll-like receptors (TLRs, Geijtenbeek &amp; Gringhuis, 2009)⁠ but also other non-TLR families of innate receptors play critical roles in innate sensing of pathogens. These are C type lectin-like receptors (Geijtenbeek &amp; Gringhuis, 2009)⁠, nucleotide-binding oligomerization domain-like receptors (Ting, Duncan, &amp; Lei, 2010)⁠, and retinoic acid-inducible gene I (RIG-I)-like receptors (Wilkins &amp; Gale, 2010)⁠.</p>
<p>These receptors, when activated, leads to increased production of inflammatory cytokines (e.g., TNFα, IFNα/β and IL-6) through the activation of transcription factors like NF-kB (Opitz, van Laak, Eitel, &amp; Suttorp, 2010)⁠⁠. This activation, coupled with the recognition of Spn by macrophages, increases the inflammatory process, further recruiting other inflammatory cells (Opitz, Van Laak, Eitel, &amp; Suttorp, 2010)⁠. Recruited neutrophils are also of extreme importance for the first host response to the pathogen. This inflammatory process in the lungs leads to a systemic response, with increased serum levels of components from the complement system and the ultimate activation of the adaptive system.</p>
<p>This activation is required to the complete clearance of Spn in humans given that it can prevent pneumonia and other invasive diseases (Martinot et al., 2014; McCool, Cate, Moy, &amp; Weiser, 2002)⁠, where both humoral (antibody) and cellular mediate responses have equal importance in controlling colonization and development of invasive disease. Despite the polysaccharide capsule being recognized as an antigen, bacterial proteins are also capable of eliciting a protective response by recruiting T lymphocytes, which induces B cell antibody response and memory B cells generation (Coutinho &amp; Möller, 1973)⁠.</p>
<p>The cellular mediated response by T lymphocytes are subdivided into Th1, Th2, Th17 and regulatory T cells, all of which are of major importance to Spn infection, especially the CD4-positive helper T cells. Th1 cells secrete important cytokines, like IFNγ, that can enhance intracellular killing by macrophages (Periselneris, 2014)⁠. Th17 cells have a fundamental role against extracellular pathogens by secreting IL-17 cytokine, increasing the recruitment of neutrophils and macrophages to the infection site. Moreover, has suggestive cross-serotype protection due to its ability in recognizing protein antigens (Moffitt et al., 2011)⁠. The anti-inflammatory role of regulatory T cells enables the control of the immune response to self and foreign particles by secreting IL-10 and reducing IFNγ production, which has been demonstrated to prevent bacteria penetration of the epithelium to reach the bloodstream, which can cause septicemia during pneumonia (Neill et al., 2012)⁠. Although the role of the Th2 responses in pneumococcal infection is not well comprehended, it is known that they secrete cytokines involved in antibody production, eosinophils activation and inhibition of phagocytes function.</p>
</div>
<div id="the-effects-of-sex-and-ageing-on-human-immunity" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> The Effects of Sex and Ageing on Human Immunity</h2>
</div>
<div id="the-experimental-human-pneumococcal-carriage-model" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> The Experimental Human Pneumococcal Carriage Model</h2>
<p>Anyone can be infected with Spn; however, not everybody can develop pneumococcal carriage: some individuals have a higher susceptibility than others (Ferreira, Jambo, &amp; Gordon, 2011)⁠. This can be due to several factors, both intrinsic as extrinsic to the host biology. Understanding what makes someone more susceptible than others to develop carriage is central to develop appropriate vaccines. Although several of those risk factors are well known, not all of them are truly understood and others remain to be uncovered. Further investigation is mandatory to uncover the essential factors as well as how they orchestrated with our immune system and Spn infection, as well when interacting with the influenza virus and vaccines.</p>
<p>To such, the Experimental Human Pneumococcal Carriage (EHPC) model ⁠was developed (Ferreira, Jambo, &amp; Gordon, 2011). It is a framework to safely explore the mechanisms and dynamics of the disease: human volunteers are inoculated intranasally with a serotype 6B strain leading to successful nasopharyngeal colonization in approximately half of the subjects (Niederman et al., 2001)⁠. The research is mainly focused on the overall host responses in the nasal mucosa to the pneumococcal carriage, specific responses of B and T cell in the lungs and blood, as well the host-pathogen interactions. Such information is obtained from new methods of mucosal nasal sampling, that are used to investigate cellular responses to carriage.</p>
<p>In this model was observed that previous colonization prevented recolonization with the same strain (Niederman et al., 2001)⁠, where all recruited subjects had detectable Immunoglobulin G (IgG) levels to Spn antigens before its inoculation. However, anti-pneumococcal IgG levels did not correlate with the success of the colonization (Niederman et al., 2001)⁠. In contrast, antibodies to the Spn surface protein (PspA) were only detected in subjects that developed carriage after challenge (Niederman et al., 2001) and correlated with prevention of successful colonization, suggesting that anti-protein antibody may prevent colonization (McCool et al., 2002)⁠. IL-17 secreting CD4+ cells specific to Spn were also found in the lung before exposure to the 6B strain, increasing their levels to 8- and 17-fold in the blood and bronchoalveolar fluid 17of volunteers successfully colonized (Neill et al., 2012)⁠.</p>
</div>
<div id="systems-vaccinology" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Systems Vaccinology</h2>
<p>The immune system can sense and control external threats through several signalling systems. Such mechanisms can confer protection by an orchestrated cascade of events originating from the infection environment to the lowest molecular levels, where the components are perturbed by the incoming signal. The perturbed components then process and integrate the signal into response, which is reversely propagated in a bottom-up fashion towards the environment. Different immune responses have their own transcriptional patterns or molecular signatures rapidly induced after the challenge. These molecular signatures correlate with and predict further protective immune responses, which means that its characterization can be a great strategy to uncover molecular mechanisms or even prospectively determine vaccine efficacy (Pulendran, Li, &amp; Nakaya, 2010)⁠.</p>
<p>Despite the apparent conceptual simplicity, there are more than 26,000 genes in our genomes and the challenge of a pathogen in the host body could perturb the expression of a substantial fraction of them (Pulendran, Li, &amp; Nakaya, 2010)⁠, a fraction that could still present extremely difficult if using traditional reductionist approaches. Systemic approaches, however, turn this problem tractable by providing us with a global picture of the biological response to such challenges (Pulendran, 2014)⁠. With new technologies for measuring the behaviour of different layers of biological systems, such as genes, molecules, cells and so forth, and coupled with the latest advances in computational and mathematical tools for dealing with such complexity, we have an unprecedented opportunity to understand the fundamental features involved in such questions (Pulendran, 2014; Pulendran, Li, &amp; Nakaya, 2010)⁠.</p>
<p>Systems biology is an interdisciplinary approach that systematically describes the complex interactions between all the parts in a biological system, to elucidate biological rules underlying the behaviour of the biological system (Pulendran, Li, &amp; Nakaya, 2010, Kitano, 2002)⁠⁠. Data are collected for different components simultaneously, representing different levels of the system under different perturbations or temporal stage. They are further integrated to generate a model that could describe or predict the behaviour of such mechanisms during different scenarios, aiming to the comprehensive understanding of biological network nature (Ideker et al., 2001; Kitano, 2002; Pulendran et al., 2010).</p>
<p>Such networks are reductive representations lower functional units of complex biological processes, such as nutrient signalling, immune response and energy production, being able to receive, compute, integrate and communicate information from the inside (genome) to the outside (environment) of a biological system (Ideker et al., 2001; Kitano, 2002; Pulendran et al., 2010)⁠. Through these models, is possible to evaluate their nature by its dynamics, robustness and plasticity upon the influence of a defiant environment.</p>
<p>This approach has two broad applications in the context of the infectious diseases, with distinct methodologies and rationales: scientific discovery and prediction of immunogenicity and efficacy of vaccines. For instance, to comprehend mechanisms of innate and adaptive immunity in various organisms (Aderem &amp; Hood, 2001; Haining et al., 2008; Haining &amp; Wherry, 2010; Kaech, Wherry, &amp; Ahmed, 2002; N. Subramanian, Torabi-Parizi, Gottschalk, Germain, &amp; Dutta, 2015; Wherry et al., 2007; Zak &amp; Aderem, 2009), including humans (Aderem &amp; Hood, 2001; N. Subramanian et al., 2015)⁠, and for identification of infectious diseases biomarkers with diagnostic purposes (Chaussabel et al., 2008; Lee et al., 2008; Otaegui et al., 2009; Ramilo et al., 2007).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="research-questions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["phd_thesis.pdf", "phd_thesis.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
